### ISSUE TITLE
If you open a select menu the scroll bar disappears

### ISSUE THREAD

**Issue by albert0815** [material-components/material-components-web#2246]
With the current version of the select menu the scroll bar disappears when you open the select. So you cannot scroll anymore on the page and also content which is aligned to center or right jumps slidly to the right.

### What MDC Web Version are you using?
v0.30.0

### What browser(s) is this bug affecting?
Tested Chrome (63.0.3239.132), Firefox (58.0.1 (64-Bit)) and IE, all are same

### What OS are you using?
Win 7

### What are the steps to reproduce the bug?
1. Open current demo of select: https://material-components-web.appspot.com/select.html 
2. Make sure you can see a scrollbar (reduce window size if you don't see one)
2. Click on the select on the top of the page ("Pick a food group").
3. The scroll bar disappeared.

As soon as the menu is opened the scroll bar disappears. I guess this is intended due to some reason but I don't understand why.
And also I think this is in fact an issue (at least a minor but still) because the content on the page I am currently working on with this (btw really beautiful) library has this problem which doesn't look good to me. Also you are not able to scroll anymore (e.g. for looking up something else on the same page which you would like to know for selecting the correct option).
Let's assume it is intended and cannot (or should not) be changed: Do you have any suggestion how I could avoid the change of position / size of other components on the same page (e.g. a card)?

### What is the expected behavior?
Scroll bar should not disappear.

### What is the actual behavior?
Scroll bar disappears.

**Comment by theenoahmason** [comment]
Just trolling the issues and saw this.

Select menu uses fixed positioning and locks the body on scroll. This means you can't scroll when the menu is open and is intentional.

Since the menu is fixed, scrolling with it open would mean the menu would stay on screen relative to the viewport. Locking the body remedies this.

I have seen the MDC web devs talking about closing the select menu/unlocking body on scroll, bit this would not make the scrollbars reappear.

**Comment by albert0815** [comment]
Thank you for your comment. That explains _from a technical point of view_ why the scrollbar disappears. The bar disappears because the body is locked and that is to workaround the menu uses fixed positioning (and would not reflect scrolling). But I don't think this makes my issue invalid. Because it does not explain why the menu uses fixed positioning. I would assume due to some technical reason which didn't allow relativ positioning. Or would there be a _more business / requirements related point of view_ explanation (where business / requirements should mean something like user experience)?

By the way: On an iOS device I noticed that the component behaves differently. There scrolling is possible but the menu does not scroll. And your comment explains why. It is fixed. And somehow I would tend to say this as well an issue. At least it is not consistent. But I would say disabling the scrolling on iOS would not fix this one :).

**Comment by kfranqueiro** [comment]
There's a longer thread on this at #879.

TL;DR this is consistent with how some (but not all) browsers behave when native select elements are opened. Others close the menu on scroll, however that poses problems when it comes to menus that themselves have scrolling in addition to the body (since as soon as the menu scrolls to the end, the browser will start scrolling the body, resulting in immediately closing the menu).

I seem to recall some iOS user agents not properly restricting scroll on body even when overflow is set to hidden, which might explain what you're seeing there. I suspect the same thing would happen with MDC Dialog, then.

As for MDC Select's menu being `position: fixed`, this is something I'm curious about as well, given that MDC Menu simply uses `absolute`.

I'm going to enter separate issues for these two things and close this, since they seem separate from the original question which already has a relevant issue.

**Comment by kmmbvnr** [comment]
Currently, to fix this, I'm using mutation observer on the body and set fixed body.style.width if scroll is locked - https://github.com/viewflow/django-material/blob/v2/material/components/dmc-scroll-lock-fix/index.js

**Comment by cyberjetx** [comment]
Regarding a MAC and scrollbar weirdness. I guess people still use them.  (Don't get upset, I'm on a macbook) 

[Chris Coyier](https://css-tricks.com/author/chriscoyier/)  wrote on on **Aug 24, 2021**
https://css-tricks.com/scrollbar-reflowing/#:~:text=The%20classic%20fixes%3A,element%20to%20prevent%20the%20scrollbar. 

Søren Birkemeyer's comment with link on August 25, 2021  
There’s so much going on in this space! [Important cross-reference](https://www.w3.org/TR/css-values-4/#viewport-relative-lengths): 

I hope something here helps and does not create noise in the discussion.

**Comment by StafnyVk** [comment]
Is this issue fixed?I noticed the same too. It still exsists in both Chrome and Bing
